tests/PipelineMicroserviceCLI/Commands/CommandTest.php

```diff
@@ -7,41 +7,89 @@
         ] );
-        $this->assertRegExp('/.*["\']?published["\']?\s?:\s?["\']?Published["\']?.*["\']?id["\']?:1,/', $commandTester->getDisplay());
+        
+//         echo $commandTester->getDisplay();
+        
+        $this->assertRegExp('/Publishing pipeline:/', $commandTester->getDisplay());
+        $this->assertRegExp('/.*["\']?published["\']?\s?:\s?["\']?Published["\']?/', $commandTester->getDisplay());
     }
     
-    public function testHidePipelineMissingBaseUrl(){
+    public function testHidePipeline(){
         $mockHandler = $this->getHttpMockHandler(["PublicPipelines.txt", "PipelineToPublish.txt"]);
         $application = new Application();
         $application->add(new HidePipeline(null,$mockHandler));
     
         $command       = $application->find('pipeline:hide');
         
+        $helper = $command->getHelper('question');
+        $helper->setInputStream($this->getInputStream("0\n y \n"));
+        
+        $commandTester = new CommandTester($command);
+        $commandTester->execute( [
+                'command' => $command->getName(),
+                "base_url" => "todo"
+        ] );
+
+//         echo $commandTester->getDisplay();
+        $this->assertRegExp('/Unpublishing pipeline:/', $commandTester->getDisplay());
+        $this->assertRegExp('/.*["\']?published["\']?\s?:\s?["\']?Hidden["\']?/', $commandTester->getDisplay());
+    }
+    
+    public function testApprovePipelineRelease(){
+        $mockHandler = $this->getHttpMockHandler(["PublicPipelines.txt", "PipelineReleaseApproved.txt"]);
+        $application = new Application();
+        $application->add(new ApprovePipelineRelease(null,$mockHandler));
+    
+        $command       = $application->find('pipeline:approve');
+    
         // Equals to a user inputting "2" and hitting ENTER
         $helper = $command->getHelper('question');
-        $helper->setInputStream($this->getInputStream("2\n y \n"));
+        $helper->setInputStream($this->getInputStream("0\n 0 \n y \n"));
+    
+        $commandTester = new CommandTester($command);
+        $commandTester->execute( [
+                'command' => $command->getName(),
+                "base_url" => "todo"
+        ] );
+//         echo $commandTester->getDisplay();
         
+        $this->assertRegExp('/.*Are you sure to approve release.*/', $commandTester->getDisplay());
+    }
+    
+    public function testDenyPipelineRelease(){
+        $mockHandler = $this->getHttpMockHandler(["PublicPipelines.txt", "PipelineReleaseDenied.txt"]);
+        $application = new Application();
+        $application->add(new DenyPipelineRelease(null,$mockHandler));
+    
+        $command       = $application->find('pipeline:deny');
+    
+        // Equals to a user inputting "2" and hitting ENTER
+        $helper = $command->getHelper('question');
+        $helper->setInputStream($this->getInputStream("0\n 2 \n y \n"));
+    
         $commandTester = new CommandTester($command);
```

Comment:
We can probably put this in a testing application that creates and gives access to CommandTester instances.

```
$testApplication->getCommandTester(TestClass::CLASS);
```